### ISSUE TITLE
Infinite recursion in httpCoerce with v2.23.0

### ISSUE THREAD

**Issue by AaronM04** [strongloop/strong-remoting#268]
Hi, so I did an npm install without upgrading my loopback version, and it bumped loopback's strong-reporting dependency from v2.22.0 to the next minor version. Now I get an infinite recursion:

```
RangeError: Maximum call stack size exceeded
    at Object.hasOwnProperty (native)
    at coerceString (/Users/[censored]/git/[censored]/node_modules/loopback/node_modules/strong-remoting/lib/http-coerce.js:56:17)
    at httpCoerce (/Users/[censored]/git/[censored]/node_modules/loopback/node_modules/strong-remoting/lib/http-coerce.js:35:13)
    at httpCoerce (/Users/[censored]/git/[censored]/node_modules/loopback/node_modules/strong-remoting/lib/http-coerce.js:31:20)
    at httpCoerce (/Users/[censored]/git/[censored]/node_modules/loopback/node_modules/strong-remoting/lib/http-coerce.js:31:20)
    at httpCoerce (/Users/[censored]/git/[censored]/node_modules/loopback/node_modules/strong-remoting/lib/http-coerce.js:31:20)
    at httpCoerce (/Users/[censored]/git/[censored]/node_modules/loopback/node_modules/strong-remoting/lib/http-coerce.js:31:20)
    at httpCoerce (/Users/[censored]/git/[censored]/node_modules/loopback/node_modules/strong-remoting/lib/http-coerce.js:31:20)
    at httpCoerce (/Users/[censored]/git/[censored]/node_modules/loopback/node_modules/strong-remoting/lib/http-coerce.js:31:20)
    at httpCoerce (/Users/[censored]/git/[censored]/node_modules/loopback/node_modules/strong-remoting/lib/http-coerce.js:31:20)
    at httpCoerce (/Users/[censored]/git/[censored]/node_modules/loopback/node_modules/strong-remoting/lib/http-coerce.js:31:20)
    at httpCoerce (/Users/[censored]/git/[censored]/node_modules/loopback/node_modules/strong-remoting/lib/http-coerce.js:31:20)
    at httpCoerce (/Users/[censored]/git/[censored]/node_modules/loopback/node_modules/strong-remoting/lib/http-coerce.js:31:20)
    at httpCoerce (/Users/[censored]/git/[censored]/node_modules/loopback/node_modules/strong-remoting/lib/http-coerce.js:31:20)
    at httpCoerce (/Users/[censored]/git/[censored]/node_modules/loopback/node_modules/strong-remoting/lib/http-coerce.js:31:20)
    at httpCoerce (/Users/[censored]/git/[censored]/node_modules/loopback/node_modules/strong-remoting/lib/http-coerce.js:31:20)
```

**Comment by AaronM04** [comment]
This is with a remote method configured as follows:
```
 user.remoteMethod "<snip>",
    http: { verb: "post", path: "/:id/<snip>/:fk" }
    accepts:[
      { arg: "id", type: "number", required: true }
      { arg: "fk", type: "number", required: true }
      { arg: "questions", type: "array[object]", required: true }
      { arg: "ctx", type: "object", required: true, http: (ctx) -> ctx }
    ]
    returns: { type: "object", root: true }
```

Could this have anything to do with the ctx argument (which is recursively defined)?

**Comment by raymondfeng** [comment]
I think `ctx` has circular references. It should be excluded from coercion.

**Comment by raymondfeng** [comment]
@STRML @bajtos We should exclude `req`, `res`, and `ctx` from coercion.

**Comment by STRML** [comment]
I agree, as I mentioned before, the patch was merged half-finished. 2.23.0 should not have been released.

**Comment by STRML** [comment]
See https://github.com/strongloop/strong-remoting/pull/265

**Comment by bajtos** [comment]
I apologise for the troubles. I am going to revert the patch that introduced this problem - see https://github.com/strongloop/strong-remoting/pull/269

**Comment by bajtos** [comment]
Should be fixed in `v2.23.1`.
